File: Study Note/Algorithm Problems/SWEA/D3/3376 파도반 수열 (pass)/SWEA_3376.py
import sys
sys.stdin = open("input.txt", "r")

arr = [1, 1, 1, 2, 2]
n = 5
while n < 100:
    arr.append(arr[n - 1] + arr[n - 5])
    n += 1

T = int(input())
for test_case in range(1, T + 1):
    N = int(input())
    print('#{} {}'.format(test_case, arr[N - 1]))

File: Study Note/Algorithm Problems/SWEA/D3/3376 파도반 수열 (pass)/SWEA_3376.c
#include <stdio.h>
int main(void)
{
	int test_case;
	int T, N, i;
	unsigned long int arr[100];
    
    for (i = 0; i < 3; i++)
        arr[i] = 1;
    for (i = 3; i < 5; i++)
        arr[i] = 2;
    for (i = 5; i < 100; i++)
        arr[i] = arr[i - 1] + arr[i - 5];

	scanf("%d", &T);

	for (test_case = 1; test_case <= T; ++test_case)
	{
		scanf("%d", &N);
		printf("#%d %lu\n", test_case, arr[N - 1]);
	}
	return 0; 
}